pkg/services/frontend/context_middleware.go

```diff
@@ -0,0 +1,41 @@
+package frontend
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/infra/tracing"
+	"github.com/grafana/grafana/pkg/services/contexthandler/ctxkey"
+	contextmodel "github.com/grafana/grafana/pkg/services/contexthandler/model"
+	"github.com/grafana/grafana/pkg/web"
+)
+
+// Minimal copy of contextHandler.Middleware for frontend-service
+// frontend-service doesn't handle authentication or know what signed in users are
+func (s *frontendService) contextMiddleware() web.Middleware {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
After the changes in https://github.com/grafana/grafana/pull/108188, I used the litmus test of "does this do I/O" to determine whether this function needs tracing spans. As it doesn't i removed them.